### PR TITLE
[Fix](Error handling with status code)

### PR DESCRIPTION
**Pull request by cb-amutha**
Added httpStatusCode to handle status code in flutter. the issue link is - https://github.com/chargebee/chargebee-flutter/issues/40

### CODE REVIEW

File: Chargebee/Classes/Result/CBError.swift [hunk 1]
```diff
@@ -29,6 +29,16 @@ extension CBError: LocalizedError {
             return errorResponse.message
         }
     }
+    public var httpStatusCode: Int {
+        switch self {
+        case .operationFailed(let errorResponse):
+            return errorResponse.httpStatusCode ?? 0
```
Comment (cb-sriramthiagarajan) [inline Chargebee/Classes/Result/CBError.swift:35]: Why are we using `0` as the fallback? 🤔

Comment (cb-amutha) [reply Chargebee/Classes/Result/CBError.swift:35]: Internally httpStatusCode is Optional value(Int? ). Just not to pass optional value to flutter(if i pass Optional value customer might ask why its Optional). Added default value of Int on the condition

Comment (cb-haripriyan) [reply Chargebee/Classes/Result/CBError.swift:35]: Can we change the param `httpStatusCode` in `CBErrorDetail` to not be optional?

Comment (cb-amutha) [reply Chargebee/Classes/Result/CBError.swift:35]: Yes, we can do that. let's check and update the PR.

Comment (cb-amutha) [reply Chargebee/Classes/Result/CBError.swift]: Done!

**Review by cb-sriramthiagarajan — APPROVED** [review]

**Pull request merged**